------------------------------------------------------------------------
r2629 | jwoithe | 2016-10-09 13:00:28 +0900 (Sun, 09 Oct 2016) | 1 line

TimestampedBuffer.cpp: move the include of math.h to the top of the file which is where most people would expect it to be.  The patch comes from a Fedora RPM source package via Simon Lewis, who suggested that without it gcc6 had trouble with the file (the nature of the problem is not known by the FFADO project at this time).  Earlier versions of gcc (for example, 4.8.2) had no trouble compiling this file without any math.h include, so the issue that some have seen with gcc6 must be due to math.h being removed from some other system-level include file at some point.
------------------------------------------------------------------------
Index: src/libutil/TimestampedBuffer.cpp
===================================================================
--- a/src/libutil/TimestampedBuffer.cpp	(revision 2628)
+++ b/src/libutil/TimestampedBuffer.cpp	(revision 2629)
@@ -32,6 +32,7 @@
 
 #include <cstdlib>
 #include <cstring>
+#include <math.h>
 
 #define DLL_PI        (3.141592653589793238)
 #define DLL_SQRT2     (1.414213562373095049)
@@ -179,8 +180,6 @@
     m_wrap_at=w;
     return true;
 }
-#include <math.h>
-
 /**
  * \brief return the effective rate
  *
